.github/workflows/qaqc_deploy.yml

```diff
@@ -4,26 +4,64 @@ on:
   workflow_dispatch:
 
 jobs:
-  deploy:
+  build_and_push:
     runs-on: ubuntu-22.04
     defaults:
       run:
         shell: bash
-      # working-directory: apps/qaqc ## currently needs to be run from root for git subtree command
-    container: 
-      image: dokku/ci-docker-image:latest
     env:
-      SSH_PRIVATE_KEY: ${{ secrets.DOKKU_SSH_KEY }}
-      DOKKU_HOST: ${{ secrets.DOKKU_HOST }}
-      APP_NAME: edm-data-engineering
-      GIT_REMOTE_URL: ssh://dokku@${{ secrets.DOKKU_HOST }}/edm-data-engineering
-      APP_REMOTE_BRANCH: main
-
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+      DOCKER_USER: ${{ secrets.DOCKER_USER }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          sparse-checkout: apps/qaqc
-      - name: Deploy
-        run: apps/qaqc/bash/deploy.sh 
+          sparse-checkout: |
+            apps/qaqc
+            bash
+            dcpy
+      - name: Build and Push
+        run: apps/qaqc/bash/build_and_publish.sh
 
+  deploy:
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+    env: 
+      QAQC_DROPLET_SSH_PRIVATE_KEY: ${{ secrets.QAQC_DROPLET_SSH_PRIVATE_KEY }}
+      QAQC_DROPLET_SSH_HOST: ${{ secrets.QAQC_DROPLET_SSH_HOST }}
+      AWS_S3_ENDPOINT: ${{ secrets.AWS_S3_ENDPOINT }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      SQL_ENGINE_EDM_DATA: ${{ secrets.SQL_ENGINE_EDM_DATA }}
+      DOCKER_IMAGE_NAME: nycplanning/qaqc:latest
+    steps:
+      - name: Setup SSH
+        run: |
+          mkdir -p ~/.ssh
+          echo "$QAQC_DROPLET_SSH_PRIVATE_KEY" > ~/.ssh/qaqc.key
+          chmod 400 ~/.ssh/qaqc.key
+          cat >> ~/.ssh/config <<EOF
+          Host qaqc
+            HostName $QAQC_DROPLET_SSH_HOST
+            User root
+            IdentityFile ~/.ssh/qaqc.key
+            StrictHostKeyChecking no
+          EOF
+      - name: Deploy
+        run: |
+          ssh qaqc "\
+            docker pull $DOCKER_IMAGE_NAME
+            pid=\"\$(docker ps -a -q -f ancestor=$DOCKER_IMAGE_NAME)\"
+            if [[ \$pid ]]; then
+              docker stop \"\$pid\"
+              docker rm \"\$pid\"
+            fi
+            docker run -p 8501:8501 -d --restart always\
+                -e AWS_S3_ENDPOINT=$AWS_S3_ENDPOINT\
+                -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY\
+                -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID\
+                -e SQL_ENGINE_EDM_DATA=$SQL_ENGINE_EDM_DATA\
+                $DOCKER_IMAGE_NAME"
```

Comment:
@fvankrieken gotcha, good to hear. I still think that we probably want to investigate using Digital Ocean Apps. Just thinking that we'd get access to all sorts of nice stuff... zero-downtime deploys, alerts, automatic restarts if the app goes down, easy subdomain management, log tailing, configuration in the GUI rather than env vars (and maybe access to secrets via DO?) and a nice CLI that would probably do most of the hairy deployment stuff with a single command. (Just remembering my time with AWS Elastic Beanstalk, and assuming DO has something similar)

Reply:
Agreed. Still think I'm going to get this merged roughly as-is for now to do some of the work I had originally intended to do on the qaqc app, but will definitely come back to this, maybe next week